modules/core/src/main/java/org/apache/ignite/internal/util/tostring/GridToStringBuilder.java

```diff
@@ -1894,6 +1899,20 @@ public static String compact(Collection<Integer> col) {
         return compact(col, i -> i + 1);
     }
 
+    /**
+     * Perform compaction of partition lists in logs.
+     * Conversion of consecutively arranged partitions to the corresponding range will be performed only if
+     * system property {@link IgniteSystemProperties#IGNITE_PRINT_PARTITION_RANGES_IN_LOGS} is set to <code>true</code>.
+     *
+     * @param col Collection of partitions.
+     * @return Compacted string representation of collection with partition numbers.
+     */
+    public static String compactPartitions(Collection<Integer> col) {
+        boolean printRanges = getBoolean(IGNITE_PRINT_PARTITION_RANGES_IN_LOGS, DFLT_IGNITE_PRINT_PARTITION_RANGES_IN_LOGS);
```

Comment:
I don't think that we should read this system property every time the `compactPartitions` is called.
Potentially it can be a bottleneck just because that `java.util.Properties` is a `Hastable` and so 'Properties.getProperty()` implemented via `Hashtable.get` which is a `synchronized` method.

Reply:
I removed property.